### ISSUE TITLE
Error/bug in AQL query: Explain: missing variable #4 while planning registers

### ISSUE THREAD

**Issue by miguel-porto** [arangodb/arangodb#1950]
Hi, I'm getting this error which appears to be a bug:

> Explain: missing variable #4 while planning registers (exception location: /usr/src/packages/BUILD/arangod/Aql/ExecutionNode.cpp:1163)

The original query is like this:
```
LET allv=(FOR v,e IN 1..100 OUTBOUND 'taxent/339816493982' ANY SYNONYM,EXISTS_IN,PART_OF
	FILTER e.nativeStatus IN ['NATIVE','NEAR_ENDEMIC']
	RETURN v)
FOR v IN allv
	LET children = (FOR v1 IN 1..10 INBOUND v belongs_to RETURN v1)
	RETURN MINUS(allv, children)
```
However, the strange thing is that if I replace the allv value by the very same documents but in a "literal way", the query works, i.e.
```
LET allv = [DOCUMENT('territory/349850448798'), DOCUMENT('territory/334166111134')]
// N.B. allv here is exactly the same object as before
FOR v IN allv
	LET children = (FOR v1 IN 1..10 INBOUND v belongs_to RETURN v1)
	RETURN MINUS(allv, children)
```
The problem also seems to have something to do with using the function MINUS, as if I just `RETURN children`, it always works.
Let me know if you can reproduce this, or I need to provide more details.
Cheers!

**Comment by Simran-B** [comment]
I reproduced the error with current devel.

    ... 'taxent/339816493982' ANY SYNONYM,EXISTS_IN,PART_OF
                                 ^__ is this a syntax error?

It also occurs with just a single edge collection (which is not also an AQL keyword) however.

Possibly a similar issue as with MERGE()? https://github.com/arangodb/arangodb/commit/1719d3c993823219944279f861565ef802443c8b

**Comment by jsteemann** [comment]
@Simran-B : the following is not a syntax error:
```
 ... 'taxent/339816493982' ANY SYNONYM,EXISTS_IN,PART_OF
                             ^__ is this a syntax error?
```
The `ANY` here specifies the traversal direction for the following collection.

**Comment by jsteemann** [comment]
Fixed in 3.0 and devel. The fix will become available in 3.0.4.

**Comment by miguel-porto** [comment]
This was fast! Will see if everything goes well with 3.0.4 then.
Thanks!
